lesson15/duygu/script.js

```diff
@@ -0,0 +1,137 @@
+const sentenceWithAWordToFind =
+	"The object-oriented nature of JavaScript is important to understand if you want to go further with your knowledge of the language and write more efficient code, therefore we've provided this module to help you.";
+const findTheJavascript=sentenceWithAWordToFind.indexOf("JavaScript");
+console.log(findTheJavascript);
+// 9. Extract a Substring
+// Extract the domain from an email address.
+const email = 'user-name@google.com';
+const startIndexOfDomain = email.indexOf('@') + 1;
+const domain = email.substring(startIndexOfDomain);
+
+console.log(domain);
+// 10. Replace Substring
+// Censor (replace 'bad' with 'good') a word in a sentence.
+const sentence = 'This is a bad example bad.';
+console.log(sentence.replace('bad', 'good'));
+
+// 11. Repeat a String (H)
+// Create a visual separator for console output. Expected result: "--------------------"
+const separator = "-".repeat(20);
+console.log(separator);
+// 12. Check if String Starts With (H)
+// Check if a file name starts with a specific prefix. Return true or false.
+const fileName = 'report_2024.pdf';
+console.log(fileName.startsWith('repo'));
+// 13. Check if String Ends With (H)
+// Check if a file name ends with a specific extension. Return true or false.
+console.log(fileName.endsWith('fdp'));
+// 14. Pad Start of String (H)
+// Pad a number with zeros using padStart() function to ensure it has at least 5 digits.
+const number = '42';
+const startsWithZero=number.padStart(5,'0');
+console.log(startsWithZero);
+// 15. Pad End of String (H)
+// Pad a text with spaces to align it to the right. Expected result: "Align     "
+const text = 'Align';
+const addSpaces=text.endsWith(5,' ')
+console.log(addSpaces);
+// 16. Split a String (H)
+// Split a sentence into words.
+const sentence2 = 'Learning JavaScript is fun';
+const spltSentences=sentence2.split(' ');
+console.log(spltSentences);
+// 17. Convert String to Array of Characters (H)
+// Convert a string into an array of its characters. Use sentence2 variable from exercise above.
+const spltWords=sentence2.split('');
+console.log(spltWords);
+
+// 18. Reverse a String (H)
+// Reverse the characters in a string. Use sentence2 variable from exercise above.
+
+/*
+I couldnt find any methods. I looked up w3schools and mdn website. 
+ı thought that ı can use for loop but not sure thats what you want me to do.
+*/
+
+
+
+// 19. Replace All Occurrences of a Substring (H)
+// Replace all instances of a word 'dog' in a text.
+const paragraph = "I think Ruth's dog is cuter than your dog!";
+const updatededVersionofParagraph=paragraph.replace(/dog/g,'cat');
+console.log(updatededVersionofParagraph);
+// 20. Extract a Substring (H)
+// Extract the price and currency from a string below using "slice" method and print it to console. Should return "9.99 USD"
+const planPriceString = 'Premium plan - 9.99 USD/month.';
+const justPrice=planPriceString.slice(15);
```

Comment:
in this exercises, calculate start and index for the slice function instead of using a hardcoded value. Only "9.99 USD" should be in the result.